### DISCUSSION TITLE
Slack based project setup

### DISCUSSION THREAD

**Discussion by rofe** [adobe/helix-home#216] [Ideas]
This is a proposal for a Slack based setup workflow. The goal is to 

## Pre-requisites:
- OneDrive or Google Drive folder shared with Helix Bot
- GH repo integrated with Helix Bot
- `fstab.yaml` with mountpoint
- dedicated Slack channel
 
## Setup flow:
1. User invites `@HelixBot`
   > /invite `@HelixBot`
1. User starts the setup wizard 
   > `@HelixBot` please help me with a new **setup**
1. Helix Bot starts thread with a setup workflow:
   > **Welcome to the Helix project setup wizard!**
   > This will only take a few minutes. I will walk you through the necessary steps. 
   > What is the URL of your GitHub repository?
   > [________________]
   > [ **Next** ]
1. User enters GH repo URL and clicks _Next_
1. > What is your production host name? (optional)
   > If you don't know it yet, you can always add it later.
   > [________________]
   > [**Back**] [ **Next** ]
1. User enters production host and clicks _Next_
1. Helix Bot displays a summary of the information provided/gathered from GitHub:
   > Here's what I gathered so far:
   > GitHub URL: `https://...`
   > Content: `https://...`
   > Preview: `https://...`
   > (Production: `https://...`)
   >
   > Click _Next_ if everything looks correct or go back to make changes.
   > [**Back**] [ **Next** ]
1. User confirms by clicking _Next_
1. Helix Bot adds the preview URL to the channel topic and generates a config file:
   > I generated the attached configuration file for you. In order to prove you are the owner of this project, I need to upload it to your content root (recommended) or GitHub repository:
   > [ CSV Attachment ]
   >
   > Click _Next_ when the upload is complete.
   > [**Back**] [ **Next** ]
1. > All set! Congratulations, your project is now set up for this channel.
   > Here's your Helix project info:
   > GitHub URL: `https://...`
   > Content: `https://...`
   > Preview: `https://...`
   > (Production: `https://...`)
   > Sidekick: `https://...`
   >
   > You can always type `@HelixBot` **info** to see this info again.
   > [ **Done** ]
1. User clicks _Done_

**Comment by tripodsan** [comment]
> What is the URL of your GitHub repository?

I don't think that the author knows, or needs to know. maybe the preview url is better?

>  I need to upload it to your content root (recommended) or GitHub repository:

we don't want to litter the content root anymore. it should be in `.helix` and I would not add the option for github, as this is an author that does the setup, and not the dev.

also, the slack bot could also upload the _empty_ config file to sharepoint and only tell the user to fill in the correct value.

**Reply by rofe** [reply]
> I don't think that the author knows, or needs to know. maybe the preview url is better?

Well, I think we may have to define the personas first: is it really the _author_ setting this up, or rather some sort of _admin_?

**Comment by rofe** [comment]
> also, the slack bot could also upload the empty config file to sharepoint and only tell the user to fill in the correct value.

👍 Ok, that would eliminate tedious manual steps like downloading, uploading and creating a folder and file. I like it.
